CMSgov/mint-app
# Easi 2258/nda acceptance

# EASI-2258

## Changes and Description

- New nda_agreeement Table
- GQL query to get if The NDA was accepted and when
- GQL mutation to accept / decline the NDA



## How to test this change

1. Scripts/dev up:backend
2. Check if you have an NDA
```gql
query NDAInfo{
  
  ndaInfo(){
      agreed
      agreedDts
    
  }
  
  
}
```
3.  Run the mutation to accept the NDA, you can change accept back and forth from `true` and `false`, not that acceptedDTS gets updated if you change from false to true.
```gql
mutation AcceptNDA 
{
    agreeToNDA(agree: true) {
      agreed
      agreedDts
    }
}
```

NOTE, you can also use this mutation without providing agree, as the default value is `true`
```gql
mutation AgreeToNDA 
{
    agreeToNDA{
      agreed
      agreedDts
    }
}
```
## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
